scipy/stats/_stats_py.py

```diff
@@ -4372,38 +4372,45 @@ def alexandergovern(*samples, nan_policy='propagate'):
     the alternative.
 
     """
-    samples = _alexandergovern_input_validation(samples, nan_policy)
-
-    if np.any([(sample == sample[0]).all() for sample in samples]):
-        msg = "An input array is constant; the statistic is not defined."
-        warnings.warn(stats.ConstantInputWarning(msg), stacklevel=2)
-        return AlexanderGovernResult(np.nan, np.nan)
```

Comment:
See " Special case: statistic is NaN when variance is zero" below.
To produce a warning, I was thinking I'd replace `means - var_w` with `_demean(means, var_w)` when gh-21068 merges.